galaxies/scripts/perf/run-galaxies-openshift.sh

```diff
@@ -0,0 +1,626 @@
+		fi
+	done
+}
+
+RESULTS_DIR_ROOT=${RESULTS_DIR_PATH}/galaxies-$(date +%Y%m%d%H%M)
+mkdir -p ${RESULTS_DIR_ROOT}
+
+#Adding 10 secs buffer to retrieve CPU and MEM info
+CPU_MEM_DURATION=`expr ${DURATION} + 10`
+
+throughputlogs=(throughput responsetime weberrorresponsetime_max stdev_resptime_max)
+podcpulogs=(cpu cpurequests cpulimits cpureq_in_p cpulimits_in_p)
+podmemlogs=(mem memusage memrequests memlimits memreq_in_p memlimit_in_p)
+clusterlogs=(c_mem c_cpu c_cpulimits c_cpurequests c_memlimits c_memrequests)
+total_logs=(${throughputlogs[@]} ${podcpulogs[@]} ${podmemlogs[@]} ${clusterlogs[@]} cpu_min cpu_max mem_min mem_max)
+
+# Download the required dependencies 
+# output: Check if the hyperfoil/wrk dependencies is already present, If not download the required dependencies to apply the load
+function load_setup(){
+	if [ ! -d "${PWD}/hyperfoil-0.13" ]; then
+		wget https://github.com/Hyperfoil/Hyperfoil/releases/download/release-0.13/hyperfoil-0.13.zip >> setup.log 2>&1
+		err_exit "Error: Could not download the dependencies"
+		unzip hyperfoil-0.13.zip >> setup.log
+	fi
+}
+
+# Run the wrk load
+# input: machine IP address, Result log file 
+# output: Run the wrk load on galaxies application and store the result in log file 
+function run_wrk_workload() {
+	# Store results in this file
+	IP_ADDR=$1
+	RESULTS_LOG=$2
+	# Run the wrk load
+	echo "Running wrk load with the following parameters" >> setup.log
+	cmd="${HYPERFOIL_DIR}/wrk2.sh --threads=${THREAD} --connections=${CONNECTIONS} --duration=${DURATION}s --rate=${REQUEST_RATE} http://${IP_ADDR}/galaxies"
+	echo "CMD = ${cmd}" >> setup.log
+	sleep 30
+	${cmd} > ${RESULTS_LOG}
+	sleep 20
+}
+
+# Run the wrk load on each instace of the application
+# input: Result directory, Type of run(warmup|measure), iteration number
+# output: call the run_wrk_workload for each application service
+function run_wrk_with_scaling()
+{	
+	RESULTS_DIR_J=$1
+	TYPE=$2
+	RUN=$3
+	svc_apis=($(oc status --namespace=${NAMESPACE} | grep "galaxies" | grep port | cut -d " " -f1 | cut -d "/" -f3))
+	load_setup
+	for svc_api  in "${svc_apis[@]}"
+	do
+		RESULT_LOG=${RESULTS_DIR_J}/wrk-${svc_api}-${TYPE}-${RUN}.log
+		run_wrk_workload ${svc_api} ${RESULT_LOG} &
+	done
+}
+
+# Parse the result log files
+# input:Type of run(warmup|measure), total number of runs, total number of iteration
+# output:Throughput log file(throghput, response time, total memory used by the pod, total cpu used by the pod, cluster memory usage in percentage,cluster cpu in percentage and web errors if any)
+function parseData() {
+	TYPE=$1
+	TOTAL_RUNS=$2
+	ITR=$3
+	echo "${TYPE} Runs" >> ${RESULTS_DIR_J}/Throughput-${itr}.log
+	for (( run=0 ; run<${TOTAL_RUNS} ;run++))
+	do
+		thrp_sum=0
+		resp_sum=0
+		wer_sum=0
+		svc_apis=($(oc status --namespace=${NAMESPACE} | grep "galaxies" | grep port | cut -d " " -f1 | cut -d "/" -f3))
+		for svc_api  in "${svc_apis[@]}"
+		do
+			RESULT_LOG=${RESULTS_DIR_P}/wrk-${svc_api}-${TYPE}-${run}.log
```

Comment:
Can you also parse the percentile details of hyperfoil/wrk2
Add --latency option to wrk2 command